WireDomain/Sources/WireDomain/Notifications/Builders/NewMessageNotificationBuilder.swift

```diff
@@ -0,0 +1,378 @@
+            teamName: teamName,
+            isMessageSilenced: isMessageSilenced,
+            conversationID: conversationID,
+            senderID: senderID.uuid,
+            selfUserID: selfUserID,
+            hidesNotificationContent: shouldHideNotification
+        )
+    }
+
+    func shouldBuildNotification() async -> Bool {
+        !context.isMessageSilenced
+    }
+
+    func buildContent() async -> UNMutableNotificationContent {
+        guard !context.hidesNotificationContent else {
+            return buildHiddenNotification()
+        }
+
+        switch message.content {
+        case .location:
+            return buildLocationNotification()
+        case .knock:
+            return buildPingNotification()
+        case .image:
+            return buildAssetNotification(ofType: .image)
+        case let .ephemeral(ephemeral):
+            return await buildEphemeralNotification(ephemeral: ephemeral)
+        case let .text(text):
+            return await buildTextNotification(text)
+        case let .composite(composite):
+            let text = composite.items.compactMap(\.text).first
+            return await buildTextNotification(text)
+        case let .asset(assetData):
+            switch assetData.original.metaData {
+            case .audio:
+                return buildAssetNotification(ofType: .audio)
+            case .video:
+                return buildAssetNotification(ofType: .video)
+            case .image:
+                return buildAssetNotification(ofType: .image)
+            default:
+                return buildAssetNotification(ofType: .fileUpload)
+            }
+        case .hidden:
+            return buildHiddenNotification()
+        default:
+            return UNMutableNotificationContent()
+        }
+    }
+
+    // MARK: - Build notifications
+
+    private func buildAssetNotification(ofType assetType: AssetType) -> UNMutableNotificationContent {
+        let content = UNMutableNotificationContent()
+        let isGroupConversation = context.isGroupConversation
+        let senderName = context.senderName
+
+        if let title = makeTitle() {
+            content.title = title
+        }
+
+        let body: NotificationBody = switch assetType {
+        case .image:
+            .newMessage(
+                .sharedPicture(senderName: isGroupConversation ? senderName : nil)
+            )
+        case .video:
+            .newMessage(
+                .sharedVideo(senderName: isGroupConversation ? senderName : nil)
+            )
+        case .audio:
+            .newMessage(
+                .sharedAudio(senderName: isGroupConversation ? senderName : nil)
+            )
+        case .fileUpload:
+            .newMessage(
+                .sharedFile(senderName: isGroupConversation ? senderName : nil)
+            )
+        }
+
+        content.body = body.make()
+        content.categoryIdentifier = makeCategory()
+        content.sound = makeSound()
```

Comment:
I think we should be more explicit so to avoid having to jump around to see what the values are, something like:

```swift
content.categoryIdentifier = NotificationCategory.unmutedConversation.rawValue
content.sound = .newMessage
```